src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.approved.txt

```diff
@@ -2351,86 +2343,72 @@ namespace Akka.Dispatch
+        public Akka.Dispatch.MessageDispatcher Dispatcher { get; }
+        public Akka.Dispatch.MessageQueues.IMessageQueue MessageQueue { get; }
+        public virtual void CleanUp() { }
         [System.Diagnostics.ConditionalAttribute("MAILBOXDEBUG")]
         public static void DebugPrint(string message, params object[] args) { }
-        public void Dispose() { }
-        protected virtual void Dispose(bool disposing) { }
-        protected abstract int GetNumberOfMessages();
-        public abstract void Post(Akka.Actor.IActorRef receiver, Akka.Actor.Envelope envelope);
-        public void Resume() { }
-        public void Resume(Akka.Dispatch.MailboxSuspendStatus reason) { }
-        protected abstract void Schedule();
+        public void Run() { }
         public virtual void SetActor(Akka.Actor.ActorCell actorCell) { }
-        public void Setup(Akka.Dispatch.MessageDispatcher dispatcher) { }
-        public void Start() { }
-        public void Suspend() { }
-        public void Suspend(Akka.Dispatch.MailboxSuspendStatus reason) { }
-    }
-    public abstract class Mailbox<TSys, TUser> : Akka.Dispatch.MessageQueueMailbox
-        where TSys : Akka.Dispatch.MessageQueues.IMessageQueue
-        where TUser : Akka.Dispatch.MessageQueues.IMessageQueue
-    {
-        protected Mailbox() { }
-        public override bool IsClosed { get; }
-        public override Akka.Dispatch.MessageQueues.IMessageQueue MessageQueue { get; }
-        protected TSys SystemMessages { get; }
-        protected TUser UserMessages { get; }
-        public override void BecomeClosed() { }
-        public override void CleanUp() { }
-        protected abstract TSys CreateSystemMessagesQueue();
-        protected abstract TUser CreateUserMessagesQueue();
-        protected override void Dispose(bool disposing) { }
-        protected override int GetNumberOfMessages() { }
-        public override void Post(Akka.Actor.IActorRef receiver, Akka.Actor.Envelope envelope) { }
-        protected override void Schedule() { }
+        protected void SetStatus(int newStatus) { }
+        protected bool UpdateStatus(int oldStatus, int newStatus) { }
     }
     public class Mailboxes
     {
         public static readonly string DefaultMailboxId;
         public static readonly string NoMailboxRequirement;
         public Mailboxes(Akka.Actor.ActorSystem system) { }
         public Akka.Actor.DeadLetterMailbox DeadLetterMailbox { get; }
-        public Akka.Dispatch.Mailbox CreateMailbox(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
         public System.Type FromConfig(string path) { }
-        public System.Type GetMailboxType(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
-        public System.Type LookupByQueueType(System.Type queueType) { }
+        public Akka.Dispatch.MailboxType GetMailboxType(Akka.Actor.Props props, Akka.Configuration.Config dispatcherConfig) { }
```

Comment:
Type checking methods for:
* Loading the correct `MailboxType` consistent with what's been requested in a HOCON declaration
* Loading the correct `MailboxType` consistent with what's been requested in a `IRequiresMessageQueue<TQueue>` declaration
* Verifying that the `MailboxType` produces the correct type of `IMessageQueue` needed to satisfy a `IRequiresMessageQueue<TQueue>` declaration